### ISSUE TITLE
Need to support the subcluster dryrun in version <= v1.8

### ISSUE THREAD

**Issue by yueyongyue** [kubevela/kubevela#6235]
**Describe the bug**
local  cluster no "networking.k8s.io/v1beta1" apiservice，test01 cluster with "networking.k8s.io/v1beta1" apiservice , dryrun run local cluster, report the following error. need to support the subcluster dryrun

run step(provider=multicluster,do=deploy): Found 1 errors. [(error encountered in cluster test01: Dispatch: pre-dispatch dryrun failed: Found 1 errors. [(cannot get object: no matches for kind "Ingress" in version "networking.k8s.io/v1beta1")])]

**KubeVela Version**

CLI Version: 1.8.2
Core Version: 1.8.2
GitRevision: git-360f69be
GolangVersion: go1.19.9

**Cluster information**

- local  Kubernetes Version [e.g. 1.26.2]
- subcluster  test01  Kubernetes Version [e.g. 1.17.3] 
- subcluster  test02  Kubernetes Version [e.g. v1.22.2]

**Comment by chivalryq** [comment]
Can you show the vela-core pod YAML?

**Comment by yueyongyue** [comment]
```yaml
---
  metadata: 
    name: "kubevela-vela-core-5bb9d6df57-xtj5c"
    generateName: "kubevela-vela-core-5bb9d6df57-"
    namespace: "vela-system"
    uid: "ce28d123-36a6-4a7f-a04b-40310c2b60c4"
    resourceVersion: "6980426"
    creationTimestamp: "2023-07-27T01:23:12Z"
    labels: 
      app.kubernetes.io/instance: "kubevela"
      app.kubernetes.io/name: "vela-core"
      pod-template-hash: "5bb9d6df57"
    annotations: 
      prometheus.io/path: "/metrics"
      prometheus.io/port: "8080"
      prometheus.io/scrape: "true"
    ownerReferences: 
      - 
        apiVersion: "apps/v1"
        kind: "ReplicaSet"
        name: "kubevela-vela-core-5bb9d6df57"
        uid: "ae6becaa-6ee8-4784-9dbc-8333427327f7"
        controller: true
        blockOwnerDeletion: true
  spec: 
    volumes: 
      - 
        name: "tls-cert-vol"
        secret: 
          secretName: "kubevela-vela-core-admission"
          defaultMode: 420
      - 
        name: "tls-cert-vol-cg"
        secret: 
          secretName: "kubevela-vela-core-cluster-gateway-tls-v2"
          defaultMode: 420
      - 
        name: "kube-api-access-9x8fb"
        projected: 
          sources: 
            - 
              serviceAccountToken: 
                expirationSeconds: 3607
                path: "token"
            - 
              configMap: 
                name: "kube-root-ca.crt"
                items: 
                  - 
                    key: "ca.crt"
                    path: "ca.crt"
            - 
              downwardAPI: 
                items: 
                  - 
                    path: "namespace"
                    fieldRef: 
                      apiVersion: "v1"
                      fieldPath: "metadata.namespace"
          defaultMode: 420
    containers: 
      - 
        name: "kubevela"
        image: "oamdev/vela-core:v1.8.2"
        args: 
          - "--metrics-addr=:8080"
          - "--enable-leader-election"
          - "--use-webhook=true"
          - "--webhook-port=9443"
          - "--webhook-cert-dir=/etc/k8s-webhook-certs"
          - "--optimize-mark-with-prob=0.1"
          - "--optimize-disable-component-revision"
          - "--health-addr=:9440"
          - "--disable-caps=rollout"
          - "--system-definition-namespace=vela-system"
          - "--application-revision-limit=20"
          - "--definition-revision-limit=20"
          - "--oam-spec-ver=v0.3"
          - "--enable-cluster-gateway"
          - "--cluster-gateway-url=kubevela-cluster-gateway-service:9443"
          - "--cluster-gateway-ca-file=/cluster-gateway-tls-cert/ca"
          - "--application-re-sync-period=5m"
          - "--concurrent-reconciles=4"
          - "--kube-api-qps=400"
          - "--kube-api-burst=600"
          - "--max-workflow-wait-backoff-time=60"
          - "--max-workflow-failed-backoff-time=300"
          - "--max-workflow-step-error-retry-times=10"
          - "--feature-gates=EnableSuspendOnFailure=false"
          - "--feature-gates=AuthenticateApplication=false"
          - "--feature-gates=LegacyComponentRevision=false"
          - "--feature-gates=GzipResourceTracker=false"
          - "--feature-gates=ZstdResourceTracker=true"
          - "--feature-gates=ApplyOnce=false"
          - "--feature-gates=MultiStageComponentApply=true"
          - "--feature-gates=GzipApplicationRevision=false"
          - "--feature-gates=ZstdApplicationRevision=true"
          - "--feature-gates=PreDispatchDryRun=true"
          - "--feature-gates=DisableBootstrapClusterInfo=false"
          - "--feature-gates=InformerCacheFilterUnnecessaryFields=true"
          - "--feature-gates=SharedDefinitionStorageForApplicationRevision=true"
          - "--feature-gates=DisableWorkflowContextConfigMapCache=true"
        ports: 
          - 
            name: "webhook-server"
            containerPort: 9443
            protocol: "TCP"
          - 
            name: "healthz"
            containerPort: 9440
            protocol: "TCP"
        resources: 
          limits: 
            cpu: "500m"
            memory: "1Gi"
          requests: 
            cpu: "50m"
            memory: "20Mi"
        volumeMounts: 
          - 
            name: "tls-cert-vol"
            readOnly: true
            mountPath: "/etc/k8s-webhook-certs"
          - 
            name: "tls-cert-vol-cg"
            readOnly: true
            mountPath: "/cluster-gateway-tls-cert"
          - 
            name: "kube-api-access-9x8fb"
            readOnly: true
            mountPath: "/var/run/secrets/kubernetes.io/serviceaccount"
        livenessProbe: 
          httpGet: 
            path: "/healthz"
            port: "healthz"
            scheme: "HTTP"
          initialDelaySeconds: 90
          timeoutSeconds: 1
          periodSeconds: 5
          successThreshold: 1
          failureThreshold: 3
        readinessProbe: 
          httpGet: 
            path: "/readyz"
            port: "healthz"
            scheme: "HTTP"
          initialDelaySeconds: 30
          timeoutSeconds: 1
          periodSeconds: 5
          successThreshold: 1
          failureThreshold: 3
        terminationMessagePath: "/dev/termination-log"
        terminationMessagePolicy: "File"
        imagePullPolicy: "IfNotPresent"
        securityContext: {}
    restartPolicy: "Always"
    terminationGracePeriodSeconds: 30
    dnsPolicy: "ClusterFirst"
    serviceAccountName: "kubevela-vela-core"
    serviceAccount: "kubevela-vela-core"
    nodeName: "10.2.5.134"
    securityContext: {}
    affinity: 
      podAntiAffinity: 
        preferredDuringSchedulingIgnoredDuringExecution: 
          - 
            weight: 100
            podAffinityTerm: 
              labelSelector: 
                matchLabels: 
                  app.kubernetes.io/instance: "kubevela"
                  app.kubernetes.io/name: "vela-core"
              topologyKey: "kubernetes.io/hostname"
    schedulerName: "default-scheduler"
    tolerations: 
      - 
        key: "node.kubernetes.io/not-ready"
        operator: "Exists"
        effect: "NoExecute"
        tolerationSeconds: 60
      - 
        key: "node.kubernetes.io/unreachable"
        operator: "Exists"
        effect: "NoExecute"
        tolerationSeconds: 60
    priority: 0
    enableServiceLinks: true
    preemptionPolicy: "PreemptLowerPriority"
  status: 
    phase: "Running"
    conditions: 
      - 
        type: "Initialized"
        status: "True"
        lastProbeTime: null
        lastTransitionTime: "2023-07-27T01:23:12Z"
      - 
        type: "Ready"
        status: "True"
        lastProbeTime: null
        lastTransitionTime: "2023-07-27T01:23:47Z"
      - 
        type: "ContainersReady"
        status: "True"
        lastProbeTime: null
        lastTransitionTime: "2023-07-27T01:23:47Z"
      - 
        type: "PodScheduled"
        status: "True"
        lastProbeTime: null
        lastTransitionTime: "2023-07-27T01:23:12Z"
    hostIP: "10.2.5.134"
    podIP: "10.3.0.2"
    podIPs: 
      - 
        ip: "10.3.0.2"
    startTime: "2023-07-27T01:23:12Z"
    containerStatuses: 
      - 
        name: "kubevela"
        state: 
          running: 
            startedAt: "2023-07-27T01:23:13Z"
        lastState: {}
        ready: true
        restartCount: 0
        image: "docker.io/oamdev/vela-core:v1.8.2"
        imageID: "docker.io/oamdev/vela-core@sha256:249737c00141e6dbdda29910e28f4b7f87f0ac49e9e51ef78c35987625a600b3"
        containerID: "containerd://3274872509581828f4311b0f51a149e178d921da0e9a05328fbbad58f13f93ce"
        started: true
    qosClass: "Burstable"
```

**Comment by chivalryq** [comment]
Hi, this problem is solved in v1.9.0+, you can try latest version of KubeVela. It discover resource endpoints from managed cluster so all cases below is supported:

1. managed clusters have legacy gvks which removed in hub cluster. 
2. managed clusters have newer gvks which hub cluster doesn't.
3. managed clusters have special CRDs which you may don't want to install in hub cluster.

It can be backported to release 1.8 but not easy, so there's no plan for backporting now.
